test-interface/src/main/scala/scala/scalanative/testinterface/TestMainBase.scala

```diff
@@ -26,9 +26,55 @@ abstract class TestMainBase {
 
   /** Actual main method of the test runner. */
   def testMain(args: Array[String]): Unit = {
-    val serverPort   = args.head.toInt
-    val clientSocket = new Socket("127.0.0.1", serverPort)
-    testRunner(Array.empty, null, clientSocket)
+    val (options, values) = args.toList.partition(_.startsWith("--"))
+    // catching Segmentation Faults and Erroneous Arithmetic Operation by default
+    // it can be disabled for debug tool compatibility
+    val catchSegfaults = !options.contains("--no-catch-fault")
+    if (catchSegfaults) {
+      signal.signal(signal.SIGSEGV, TestMainBase.faultHandler)
+      signal.signal(signal.SIGFPE, TestMainBase.faultHandler)
+    }
+    values match {
+      case "run-test" :: className :: Nil =>
+        runSingleTest(className)
+      case serverPort :: _ =>
+        val clientSocket = new Socket("127.0.0.1", serverPort.toInt)
+        testRunner(Array.empty, null, clientSocket)
+    }
+  }
+
+  private def runSingleTest(value: String): Unit = {
+    val runner = frameworks(0).runner(Array.empty,
+                                      Array.empty,
+                                      new PreloadedClassLoader(tests))
+    val taskDef = new TaskDef(
+      value,
+      DeserializedSubclassFingerprint(isModule = true,
+                                      "tests.Suite",
+                                      requireNoArgConstructor = false),
+      false,
+      Array(new SuiteSelector))
+    val Array(task: Task) = runner.tasks(Array(taskDef))
+    val logger = new Logger {
+      def debug(msg: String): Unit = Console.err.println("DEBUG: " + msg)
+
+      def error(msg: String): Unit = Console.err.println("ERROR: " + msg)
+
+      val ansiCodesSupported = true
+
+      def warn(msg: String): Unit = Console.err.println("WARN: " + msg)
+
+      def trace(t: Throwable): Unit = {
+        Console.err.println("TRACE:")
+        t.printStackTrace()
+      }
+
+      def info(msg: String): Unit = Console.err.println("INFO: " + msg)
+    }
+    val eventHandler = new EventHandler {
+      def handle(event: SbtEvent): Unit = {}
+    }
+    task.execute(eventHandler, Array(logger))
```

Comment:
The goal is to run a single test directly using the binary that sbt generates with `test:nativeLink`? Unfortunately, this will only work with our own tests, even though the option will be available to everyone.

I was thinking about the following approach:
 - Add a `lazy val nativeDebug = settingKey[Boolean]("Enable debugging of tests with LLDB")` to the sbt plugin
 - When running the tests:
    - if `nativeDebug in Test` is `false`, the tests are run exactly as they are today.
    - if `nativeDebug in Test` is `true`, then sbt **does not** start the test process itself. Instead, it prints out a command line that users should copy-paste in their terminal and waits for the user to press `return`:
> Copy-paste the following command to open the tests in LLDB, start the program, and then press `return` in this window to continue:
> $ lldb target/scala-2.11/classes/test-out 63559

Reply:
It should probably say that another debugger can be used, but the first argument to the test program should be the given port number.

Reply:
Yes, that was what I was experimenting with too. A port between `1024` and `65535`, otherwise you get an error.

Reply:
1. Sockets will likely cause problems when using breakpoints in the debugger. The debugger stops at a break point for a while. The JVM part is waiting to receive a message, then timeouts. All further tests fail with `SocketClosed`. It should be better to provide a command for running a single test directly. Currently I have some parameters hardcoded, but those could also be passed via command line.
2. A direct command would be more user friendly than a flag. For example, `testCommands` could take the same arguments as `testOnly` and produce a list of commands to run - 1 for each test suite. 
```bash
sbt "; set nativeDebug := true; testOnly java.lang.ThreadSuite"
```
vs
```bash
sbt testCommands java.lang.ThreadSuite
```